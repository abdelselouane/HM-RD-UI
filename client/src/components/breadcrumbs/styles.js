import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
