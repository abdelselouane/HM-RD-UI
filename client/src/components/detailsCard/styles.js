import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
