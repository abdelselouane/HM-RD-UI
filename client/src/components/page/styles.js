import styled, { css } from 'styled-components';

const defaultGrid = css`
  display: grid;
  grid-template-rows: 70px
                      48px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 72px);
  grid-template-columns: 20px 1fr repeat(4, 3fr) 1fr 20px;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . . . . . . .";
`;

const lowResolutionGrid = css`
  display: grid;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". . . . . . . .";
  grid-template-rows: 70px
                      24px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 48px);
  grid-template-columns: 10px 1fr repeat(4, 3fr) 1fr 10px;
`;

export const GridLayout = styled.div`
  background-color: ${({ theme }) => theme.black12};
  box-sizing: border-box;
  color: ${({ theme }) => theme.blackA11};
  height: 100%;  
  ${props => (props.enableScroll ? 'overflow-x: scroll;' : 'overflow: hidden;')}
  
  ${() => defaultGrid}
  @media only screen and (max-width: 1024px) {
    ${() => lowResolutionGrid}
  }
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  grid-area: carea;
  justify-content: center;
  ${({ hideBoxShadow }) => (hideBoxShadow ? '' : 'box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);')}
`;
