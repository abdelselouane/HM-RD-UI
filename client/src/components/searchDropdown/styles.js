import styled from 'styled-components';

const StyledSearchDropdown = styled.div`
  display: block;
  width: 700px;
  position: absolute;
  top: 70%;
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);
  z-index: 5;

  .scrollable-table {
    overflow-y: hidden;
  }

  .standard-table tbody tr {
    td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &:hover td {
      background-color: ${({ theme }) => theme.black10} !important;
    }
  }

  .table-body.alternating-rows {
    width: 100%;
    
    tr:nth-child(2n + 1),
    tr:nth-child(2n) {
      background-color: ${({ theme }) => theme.black10};
    }

    tr:hover td,
    tr:focus td {
      background-color: ${({ theme }) => theme.black12} !important;
    }
    
    tr td {
      border: none;
    }
  }
`;

export default StyledSearchDropdown;
