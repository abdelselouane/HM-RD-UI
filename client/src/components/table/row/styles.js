import styled from 'styled-components';

const StyledTd = styled.td`
  ${props => {
    if (props.tdProps
      && props.tdProps.color
      && Object.keys(props.theme).includes(props.tdProps.color)) {
      return {
        ...props.tdProps,
        color: `${props.theme[props.tdProps.color]} !important`
      };
    }
    if (props.disableRow) {
      return {
        ...props.tdProps,
        color: `${props.theme.black14} !important`
      };
    }
    return props.tdProps;
  }}
`;

export default StyledTd;
