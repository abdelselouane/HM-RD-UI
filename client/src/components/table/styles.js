import styled, { css } from 'styled-components';

const expandableTableCss = css`
  &.standard-table > thead > tr > th {
    background-color: ${({ theme }) => theme.white} !important; 
    &.sortable {
      &:after {
        position: static !important;
        padding-left: 10px;
      }
    }  
  }

  &.standard-table > tbody > tr {
    > td {
      border-bottom: 1px solid ${({ theme }) => theme.black14};
    }

    &:nth-child(4n + 1):hover > td,
    &:nth-child(4n + 2):hover > td,
    &:nth-child(4n + 3):hover > td,
    &:nth-child(4n + 4):hover > td {
      background-color: unset !important;
    }

    &:nth-child(4n + 1) td,
    &:nth-child(4n + 2) td {
      background-color: ${({ theme }) => theme.black11} !important;
    }

    &:nth-child(4n + 3) td,
    &:nth-child(4n + 4) td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &[id^='expandable'] {
      &:nth-child(4n + 1):hover,
      &:nth-child(4n + 1):hover > td,
      &:nth-child(4n + 2):hover,
      &:nth-child(4n + 2):hover > td {
        background-color: ${({ theme }) => theme.black11} !important;
      }

      &:nth-child(4n + 3):hover,
      &:nth-child(4n + 3):hover > td,
      &:nth-child(4n + 4):hover,
      &:nth-child(4n + 4):hover > td {
        background-color: ${({ theme }) => theme.white} !important;
      }
    }
  }
`;

const clearOnHoverCss = css`
  &.standard-table > thead > tr:hover,
  &.standard-table > thead > tr:hover > th {
    background-color: inherit !important;
  }
`;

export const StyledTable = styled.table`
  background-color: unset !important;
  ${({ isExpandable }) => isExpandable && expandableTableCss} ${({ clearOnHover }) =>
  clearOnHover && clearOnHoverCss};
`;

export const StyledTh = styled.th`
  ${props => {
    if (
      props.thProps &&
      props.thProps.color &&
      Object.keys(props.theme).includes(props.thProps.color)
    ) {
      return {
        ...props.thProps,
        color: `${props.theme[props.thProps.color]} !important`
      };
    }
    return props.thProps;
  }};
`;
