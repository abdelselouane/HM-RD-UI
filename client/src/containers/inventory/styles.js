import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background-color: ${({ theme }) => theme.white};
box-sizing: border-box;
display: flex;
flex-direction: column;
justify-content: flex-start;
height: 100%;
overflow-y: hidden;
width: 100%;


.inventory-tabs {
  height: 100%;
}

.tabs span {
  &:focus {
    outline: none !important;
  }
  
  &[role=tab]:first-child {
    border-left: none;
  }
}

[role=tabpanel] {
    display: flex;
    flex-direction: column;
    height: 95%;
    overflow: hidden;
    padding: 20px 0px;

    .text-input-container.search {
      margin: 15px 15px;
      width: 65%;

      &.dropdown {
        width: 75%;
        margin: 15px 0px;
      }
    }
}

.search-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  button {
    flex: 0.15;
  }
}

.text-input-container .input-field {
  margin-top: 0em;
}

.edit-col .button {
  margin: 0px;
}

.text-input-container.search {
  flex: 0.5;

  input {
    margin-top: -6px;
  }

  .icon_search {
    bottom: 0;
    color: ${({ theme }) => theme.orange};
    font-size: 30px;
    padding: 5px 10px;
    position: absolute;
    right: 0;
    z-index: 1;
  }
}
`;

export default Container;
