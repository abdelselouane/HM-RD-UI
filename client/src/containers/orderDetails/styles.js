import styled from 'styled-components';
import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;

  h2 {
    display: inline-block;
    margin-bottom: 0px;
  }
`;

export const PurchaseOrderList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 20px;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
`;
