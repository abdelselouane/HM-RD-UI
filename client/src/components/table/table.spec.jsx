/* global beforeAll, describe, expect, it, jest */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Table from './table';
import Row from './row';
import * as Styles from './styles';

describe('<Table />', () => {
  let wrapper;
  let props;

  beforeAll(() => {
    props = {
      columns: [
        {
          name: 'Mock Column',
          key: 'mockColumn',
          sortable: true,
          orderBy: {
            className: 'ascending',
            key: 'mockColumn'
          }
        }
      ],
      rows: [
        {
          key: 'abc-1234',
          selected: false,
          mockColumn: 'abc-1234'
        }
      ],
      handleRowClick: jest.fn(),
      handleSortClick: jest.fn()
    };

    wrapper = shallow(<Table {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ isAlternating: false });
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ isExpandable: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render styles properly', () => {
    Object.values(Styles).forEach(Component => {
      expect(toJson(shallow(<Component />))).toMatchSnapshot();
    });
    expect(
      toJson(shallow(<Styles.StyledTh tdProps={{ color: 'orange' }} theme={{ orange: '#000' }} />))
    ).toMatchSnapshot();
  });

  it('should call handleSortClick method when column header is clicked', () => {
    expect(wrapper.find('.standard-table thead tr .sortable')).toHaveLength(1);
    const spy = jest.spyOn(props, 'handleSortClick');
    wrapper.find('.standard-table thead tr .sortable').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('should render header if showHeader set to true', () => {
    wrapper.setProps({ showHeader: true });
    expect(wrapper.find('.table-header thead')).toHaveLength(1);
  });

  it('should render header if showHeader set to false', () => {
    wrapper.setProps({ showHeader: false });
    expect(wrapper.find('.table-header thead')).toHaveLength(0);
  });

  it('should render checkbox column if showCheckbox set to true', () => {
    wrapper.setProps({ showHeader: true, showCheckbox: true });
    expect(wrapper.find('.table-header .checkbox-col')).toHaveLength(1);
  });

  it('should render checkbox column if showCheckbox set to false', () => {
    wrapper.setProps({ showHeader: true, showCheckbox: false });
    expect(wrapper.find('.table-header .checkbox-col')).toHaveLength(0);
  });

  it('should set the table header classname if classname is defined', () => {
    wrapper.setProps({ showHeader: true, className: 'mock-class' });
    expect(wrapper.find('.table-header.mock-class')).toHaveLength(1);
  });

  it('should set the th classname', () => {
    wrapper.setProps({ showHeader: true });
    expect(wrapper.find(Styles.StyledTh)).toHaveLength(1);
  });

  it('should set the scrollable-table style if showResults is false', () => {
    wrapper.setProps({ showResults: true });
    expect(wrapper.find('.scrollable-table').prop('style')).toMatchObject({});
  });

  it('should set the scrollable-table style if showResults is false', () => {
    wrapper.setProps({ showResults: false });
    expect(wrapper.find('.scrollable-table').prop('style')).toMatchObject({
      overflowY: 'hidden'
    });
  });

  it('should set the table body classname if classname is defined', () => {
    wrapper.setProps({ className: 'mock-class' });
    expect(wrapper.find('.table-body.mock-class')).toHaveLength(1);
  });

  it('should render no results table row if showResults is false', () => {
    wrapper.setProps({ showResults: false });
    expect(wrapper.find('tr td.no-results')).toHaveLength(1);
  });

  it('should render Row component if showResults is true', () => {
    wrapper.setProps({ showResults: true });
    expect(wrapper.find(Row)).toHaveLength(1);
  });

  it('should not render Row component if showResults is true and rows are empty', () => {
    wrapper.setProps({ showResults: true, rows: [] });
    expect(wrapper.find(Row)).toHaveLength(0);
  });
});
