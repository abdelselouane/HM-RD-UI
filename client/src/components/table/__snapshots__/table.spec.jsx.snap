// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render properly 1`] = `
<Fragment>
  <styled.table
    className="standard-table table-header "
    clearOnHover={false}
    isAlternating={true}
    isExpandable={false}
    numColumns={1}
  >
    <thead>
      <tr>
        <styled.th
          className="mock-column-col sortable ascending"
          key="mock-column"
          onClick={[Function]}
          thProps={Object {}}
        >
          Mock Column
        </styled.th>
      </tr>
    </thead>
  </styled.table>
  <div
    className="scrollable-table"
    style={Object {}}
  >
    <styled.table
      className="standard-table table-body  alternating-rows"
      clearOnHover={false}
      isAlternating={true}
      isExpandable={false}
      numColumns={1}
    >
      <tbody>
        <TableRow
          columns={
            Array [
              Object {
                "key": "mockColumn",
                "name": "Mock Column",
                "orderBy": Object {
                  "className": "ascending",
                  "key": "mockColumn",
                },
                "sortable": true,
              },
            ]
          }
          key="row abc-1234"
          onClick={[MockFunction]}
          row={
            Object {
              "key": "abc-1234",
              "mockColumn": "abc-1234",
              "selected": false,
            }
          }
          showCheckbox={false}
        />
      </tbody>
    </styled.table>
  </div>
</Fragment>
`;

exports[`<Table /> should render properly 2`] = `
<Fragment>
  <styled.table
    className="standard-table table-header "
    clearOnHover={false}
    isAlternating={false}
    isExpandable={false}
    numColumns={1}
  >
    <thead>
      <tr>
        <styled.th
          className="mock-column-col sortable ascending"
          key="mock-column"
          onClick={[Function]}
          thProps={Object {}}
        >
          Mock Column
        </styled.th>
      </tr>
    </thead>
  </styled.table>
  <div
    className="scrollable-table"
    style={Object {}}
  >
    <styled.table
      className="standard-table table-body  "
      clearOnHover={false}
      isAlternating={false}
      isExpandable={false}
      numColumns={1}
    >
      <tbody>
        <TableRow
          columns={
            Array [
              Object {
                "key": "mockColumn",
                "name": "Mock Column",
                "orderBy": Object {
                  "className": "ascending",
                  "key": "mockColumn",
                },
                "sortable": true,
              },
            ]
          }
          key="row abc-1234"
          onClick={[MockFunction]}
          row={
            Object {
              "key": "abc-1234",
              "mockColumn": "abc-1234",
              "selected": false,
            }
          }
          showCheckbox={false}
        />
      </tbody>
    </styled.table>
  </div>
</Fragment>
`;

exports[`<Table /> should render properly 3`] = `
<Fragment>
  <styled.table
    className="standard-table table-header "
    clearOnHover={false}
    isAlternating={false}
    isExpandable={true}
    numColumns={1}
  >
    <thead>
      <tr>
        <styled.th
          className="mock-column-col sortable ascending"
          key="mock-column"
          onClick={[Function]}
          thProps={Object {}}
        >
          Mock Column
        </styled.th>
      </tr>
    </thead>
  </styled.table>
  <div
    className="scrollable-table"
    style={Object {}}
  >
    <styled.table
      className="standard-table table-body  "
      clearOnHover={false}
      isAlternating={false}
      isExpandable={true}
      numColumns={1}
    >
      <tbody>
        <TableRow
          columns={
            Array [
              Object {
                "key": "mockColumn",
                "name": "Mock Column",
                "orderBy": Object {
                  "className": "ascending",
                  "key": "mockColumn",
                },
                "sortable": true,
              },
            ]
          }
          onClick={[MockFunction]}
          row={
            Object {
              "key": "abc-1234",
              "mockColumn": "abc-1234",
              "selected": false,
            }
          }
          showCheckbox={false}
        />
        <tr
          id="expandable-abc-1234"
        />
      </tbody>
    </styled.table>
  </div>
</Fragment>
`;

exports[`<Table /> should render styles properly 1`] = `
.c0 {
  background-color: unset !important;
}

<table
  className="c0"
/>
`;

exports[`<Table /> should render styles properly 2`] = `
<th
  className=""
/>
`;

exports[`<Table /> should render styles properly 3`] = `
<th
  className=""
/>
`;
